Mathlib/Geometry/Manifold/ContMDiff.lean

```diff
@@ -2030,6 +2030,83 @@ nonrec theorem Smooth.smul {f : M → 𝕜} {g : M → V} (hf : Smooth I 𝓘(
   hf.smul hg
 #align smooth.smul Smooth.smul
 
+/-! ### Smoothness of an open embedding and its inverse -/
+
+variable (I)
+lemma contMDiff_openEmbedding
+    {M : Type*} [TopologicalSpace M] [Nonempty M]
+  {e : M → H} (h : OpenEmbedding e) {n : WithTop ℕ} :
+  @ContMDiff _ _ _ _ _ _ _ I _ _ h.singletonChartedSpace _ _ _ _ _ I _ _ _ n e := by
+haveI := h.singleton_smoothManifoldWithCorners I
+rw [@contMDiff_iff _ _ _ _ _ _ _ _ _ _ h.singletonChartedSpace]
+use h.continuous
+intros x y -- show the function is actually the identity on the range of I ∘ e
+apply ContDiffOn.congr contDiffOn_id
+intros z hz -- factorise into the chart (=e) and the model (=id)
+rw [extChartAt_coe, @extChartAt_coe_symm _ _ _ _ _ _ _ _ _ _ _ h.singletonChartedSpace,
+  chartAt_self_eq]
+repeat rw [Function.comp_apply]
+rw [LocalHomeomorph.refl_apply, id.def, LocalHomeomorph.singletonChartedSpace_chartAt_eq,
+  OpenEmbedding.toLocalHomeomorph_right_inv e]
```

Comment:
Here (and at a few more places in the proof) you have another opportunity to use dot notation, if you want :-)